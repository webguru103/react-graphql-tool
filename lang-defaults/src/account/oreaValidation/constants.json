[
  {
    "id": "cannot-reach-title",
    "defaultMessage": "We couldn't reach OREA"
  },
  {
    "id": "make-request-msg",
    "defaultMessage": "In order to continue,\n    we need to verify your identity with OREA.\n    Please click the button below to verify\n    your identity."
  },
  {
    "id": "first-login-msg-accent",
    "defaultMessage": "In order to use OREA forms,"
  },
  {
    "id": "first-login-msg",
    "defaultMessage": "you need to verify your identity with OREA.\n    Please click the button below to verify\n    your identity."
  },
  {
    "id": "make-request-title",
    "defaultMessage": "Identity verification"
  },
  {
    "id": "cannot-reach-msg",
    "defaultMessage": "Sorry, but we were not able to reach\n    OREA at this moment. You can go back and\n    try again, or verify later in your account settings."
  },
  {
    "id": "do-it-later",
    "defaultMessage": "Do It Later"
  },
  {
    "id": "try-again",
    "defaultMessage": "Try Again"
  },
  {
    "id": "do-it-later-msg",
    "defaultMessage": "You can validate with OREA anytime in the Account Settings page."
  },
  {
    "id": "ok-button",
    "defaultMessage": "OK"
  },
  {
    "id": "cancel-button",
    "defaultMessage": "Cancel"
  },
  {
    "id": "do-it-later-button",
    "defaultMessage": "I'll do it later."
  },
  {
    "id": "go-to-orea",
    "defaultMessage": "Go to OREA validation page"
  },
  {
    "id": "ver-failure-msg",
    "defaultMessage": "Sorry, but we were not able to verify your identity with OREA. Please check your information and try again."
  },
  {
    "id": "ver-failure-title",
    "defaultMessage": "Verification failed"
  },
  {
    "id": "ver-sucess-msg",
    "defaultMessage": "Your identity has been successfully verified through OREA."
  },
  {
    "id": "ver-sucess-title",
    "defaultMessage": "OREA verificiation successful"
  },
  {
    "id": "done-button",
    "defaultMessage": "Done"
  }
]