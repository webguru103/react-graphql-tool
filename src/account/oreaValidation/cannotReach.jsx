import React from 'react';
import { withStyles } from '@material-ui/core/styles';

import Button from '../../shared/button/button';
import styles from './cannotReach.styles';
import c from './constants';
import IllSmallLock from '../shared/icons/ill_small_lock.svg';

type PropType = {
  classes: Object,
  changeView: Function,
}

export const CannotReachC = ({ classes, changeView }: PropType) => (
  <React.Fragment>
    <p className={classes.screenTitle} data-testid="cannot-reach">
      {c.CANNOT_REACH_TITLE}
    </p>
    <div className={classes.notice}>
      <div className={classes.noticeImageContainer}>
        <IllSmallLock style={{ width: '80px', transform: 'translate(-18px, -12px)' }} />
      </div>
      <div className={classes.noticeText}>
        {c.CANNOT_REACH_MSG}
      </div>
    </div>
    <div className={classes.controlButtons}>
      <Button
        testId="change-view-do-it-later"
        classes={{ button: classes.controlButton }}
        text={c.DO_IT_LATER}
        onClick={() => changeView('do-it-later')}
        secondary
      />
      <Button
        testId="change-view-make-request"
        classes={{ button: classes.controlButton }}
        text={c.TRY_AGAIN}
        onClick={() => changeView('make-request')}
      />
    </div>
  </React.Fragment>
);

export default withStyles(styles)(CannotReachC);
