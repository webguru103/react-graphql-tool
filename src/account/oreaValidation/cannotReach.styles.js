import sharedCss from '../shared/shared.styles';

export default (theme: Object) => ({
  screenContainer: sharedCss.screenContainer,
  screenTitle: sharedCss.screenTitle(theme),
  button: {
    textTransform: 'none',
    width: '100%',
    marginTop: 30,
    fontSize: 16,
    fontWeight: 300,
    height: 38,
  },
  controlButtons: {
    ...sharedCss.controlButtons,
  },
  controlButton: sharedCss.controlButton,
  notice: {
    display: 'flex',
    flexFlow: 'row nowrap',
  },
  noticeText: {
    fontSize: 16,
    color: theme.palette.neutrals.mainText,
  },
  noticeImageContainer: {
    minWidth: 70,
    minHeight: 70,
  },
  noticeImage: {
    width: '120%',
    height: '120%',
    transform: 'translate(-18px, -12px)',
  },
});
