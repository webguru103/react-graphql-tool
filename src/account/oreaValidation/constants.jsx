import React from 'react';
import { FormattedMessage } from 'react-intl';

const DEFAULT_PAGE_SIZE = 10;

const ENVIRONMENTS = {
  DEV: 'DEV',
  SIT: 'SIT',
  QA: 'QA',
  PROD: 'PROD',
};

export default {
  DEFAULT_PAGE_SIZE,
  ENVIRONMENTS,
  CANNOT_REACH_TITLE: <FormattedMessage id="cannot-reach-title" defaultMessage="We couldn&#39;t reach OREA" />,
  MAKE_REQUEST_MSG: <FormattedMessage
    id="make-request-msg"
    defaultMessage="In order to continue,
    we need to verify your identity with OREA.
    Please click the button below to verify
    your identity."
  />,
  FIRST_LOGIN_MSG_ACCENT: <FormattedMessage
    id="first-login-msg-accent"
    defaultMessage="In order to use OREA forms, "
  />,
  FIRST_LOGIN_MSG: <FormattedMessage
    id="first-login-msg"
    defaultMessage="you need to verify your identity with OREA.
    Please click the button below to verify
    your identity."
  />,
  MAKE_REQUEST_TITLE: <FormattedMessage id="make-request-title" defaultMessage="Identity verification" />,
  CANNOT_REACH_MSG: <FormattedMessage
    id="cannot-reach-msg"
    defaultMessage="Sorry, but we were not able to reach
    OREA at this moment. You can go back and
    try again, or verify later in your account settings."
  />,
  DO_IT_LATER: <FormattedMessage id="do-it-later" defaultMessage="Do It Later" />,
  TRY_AGAIN: <FormattedMessage id="try-again" defaultMessage="Try Again" />,
  DO_IT_LATER_MSG: <FormattedMessage id="do-it-later-msg" defaultMessage="You can validate with OREA anytime in the Account Settings page." />,
  OK: <FormattedMessage id="ok-button" defaultMessage="OK" />,
  CANCEL: <FormattedMessage id="cancel-button" defaultMessage="Cancel" />,
  ILL_DO_IT_LATER: <FormattedMessage id="do-it-later-button" defaultMessage="I'll do it later." />,
  GO_TO_OREA: <FormattedMessage id="go-to-orea" defaultMessage="Go to OREA validation page" />,
  VER_FAILURE_MSG: <FormattedMessage
    id="ver-failure-msg"
    defaultMessage="Sorry, but we were not able to verify your identity with OREA. Please check your information and try again."
  />,
  VER_FAILURE_TITLE: <FormattedMessage id="ver-failure-title" defaultMessage="Verification failed" />,
  VER_SUCCESS_MSG: <FormattedMessage id="ver-sucess-msg" defaultMessage="Your identity has been successfully verified through OREA." />,
  VER_SUCCESS_TITLE: <FormattedMessage id="ver-sucess-title" defaultMessage="OREA verificiation successful" />,
  DONE: <FormattedMessage id="done-button" defaultMessage="Done" />,
};
