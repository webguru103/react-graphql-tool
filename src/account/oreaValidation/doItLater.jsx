import React from 'react';
import { withStyles } from '@material-ui/core/styles';

import Button from '../../shared/button/button';
import styles from './doItLater.styles';
import c from './constants';

type PropType = {
  classes: Object,
  closeDialog: Function,
}

export const DoItLaterC = ({ classes, closeDialog }: PropType) => (
  <React.Fragment>
    <div className={classes.notice} data-testid="do-it-later">
      <div className={classes.noticeText}>
        {c.DO_IT_LATER_MSG}
      </div>
    </div>
    <div className={classes.controlButtons}>
      <Button
        testId="close-dialog"
        classes={{ button: classes.controlButton }}
        text={c.OK}
        onClick={closeDialog}
        secondary
      />
    </div>
  </React.Fragment>
);

export default withStyles(styles)(DoItLaterC);
