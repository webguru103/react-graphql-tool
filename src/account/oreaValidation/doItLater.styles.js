import sharedCss from '../shared/shared.styles';

export default (theme: Object) => ({
  noticeText: sharedCss.noticeText(theme),
  controlButtons: sharedCss.controlButtons,
  controlButton: sharedCss.controlButton,
  notice: {
    ...sharedCss.notice,
    marginTop: 26,
  },
});
