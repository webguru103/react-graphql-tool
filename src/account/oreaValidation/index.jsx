import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import type { Match } from 'react-router';
import uuidv1 from 'uuid/v1';
import { compose, pro } from '../../utility';
import styles from './index.styles';
import MakeRequest from './makeRequest';
import CannotReach from './cannotReach';
import DoItLater from './doItLater';
import VerificationFailed from './verificationFailed';
import VerificationSuccess from './verificationSuccess';
import { withSendOREAGuid } from '../api/accountService';
import { withAppUser } from '../../shared/authorization/userConsumer';
import { OREA_VALIDATION_ENDPOINT } from '../../configurations';
import type { UserType } from '../../flowTypes';

type PropType = {
  classes: Object,
  closeDialog: Function,
  match: Match,
  initialDialog?: string,
  sendOreaGuid: Function,
  user: UserType,
  firstLogin: boolean,
}

type StateType = {
  view: string,
}

export class OREAValidationC extends React.Component<PropType, StateType> {

  static defaultProps = {
    initialDialog: 'make-request',
  };

  state = {
    view: this.props.initialDialog ? this.props.initialDialog : 'make-request',
  };

  fetchOREA = () => {
    // Test if page is live before sending POST request.
    fetch('https://www.orea.com/member-validation', { mode: 'no-cors' })
      .then(() => {
        // If page responds, assume page is live.
        // No information is returned with no-cors request to OREA.
        this.sendGuid();
      })
      .catch(() => {
        // If page cannot be fetched, send to 'cannot reach OREA' page.
        this.changeView('cannot-reach');
      });
  };

  sendGuid = async () => {
    // Send GUID to user object in our DB. Only current user has permissions to store it there.
    // Then the guid will be sent to OREA in oreaRequest POST form request, then sent back from OREA to orion.
    // When it comes back from OREA along with status of OREA membership, we can compare it to the one stored on user object.
    const guid = uuidv1();
    // TODO: Placeholder ID used - update with current user's ID (and also validate access to updateUser with JWT)
    const [err, res] = await pro(this.props.sendOreaGuid({ id: this.props.user.id, guid }));

    if (err || !res) {
      this.changeView('cannot-reach');
      return;
    }

    // Make the OREA POST request with stored guid and user ID
    this.oreaRequest(res.data.updateUser.id, guid);
  };

  testOREADown = () => {
    this.changeView('cannot-reach');
  };

  oreaRequest = (userId: string, guid: string) => {
    const { match } = this.props;

    const formData = {
      Guid: guid,
      ReturnURL: `${OREA_VALIDATION_ENDPOINT}?redirectPath=${match.path}&userId=${userId}`,
      SecurityKey: 'orea#4498',
      PartnerName: 'DealTap',
    };

    const form = document.createElement('form');
    form.style.visibility = 'hidden';
    form.method = 'post';
    form.action = 'https://www.orea.com/member-validation';
    Object.keys(formData).forEach((item) => {
      const input = document.createElement('input');
      input.style.visibility = 'hidden';
      input.name = item;
      input.value = formData[item];
      form.appendChild(input);
    });
    if (document.body) {
      document.body.appendChild(form);
    }
    form.submit();
  };

  changeView = (view: string) => {
    this.setState({
      view,
    });
  };

  renderView = (view: string) => {
    switch (view) {
      case 'make-request':
        return (
          <MakeRequest
            firstLogin={this.props.firstLogin}
            fetchOREA={this.fetchOREA}
            testOREADown={this.testOREADown}
            closeDialog={this.props.closeDialog}
            changeView={this.changeView}
          />
        );
      case 'cannot-reach':
        return <CannotReach changeView={this.changeView} />;
      case 'do-it-later':
        return <DoItLater closeDialog={this.props.closeDialog} />;
      case 'verification-failed':
        return <VerificationFailed changeView={this.changeView} />;
      case 'verification-success':
        return <VerificationSuccess closeDialog={this.props.closeDialog} />;
      default:
        return null;
    }
  };

  render() {
    const { classes } = this.props;
    const { view } = this.state;

    return (
      <div className={classes.dialogContainer}>
        {this.renderView(view)}
      </div>
    );
  }

}

export default compose(
  withStyles(styles),
  withSendOREAGuid,
  withAppUser,
)(OREAValidationC);
