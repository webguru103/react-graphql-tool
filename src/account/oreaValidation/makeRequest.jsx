import React from 'react';
import { withStyles } from '@material-ui/core/styles';

import { ENVIRONMENT } from '../../configurations';
import Button from '../../shared/button/button';
import styles from './makeRequest.styles';
import c from './constants';
import IllSmallAccess from '../shared/icons/ill_small_access.svg';

type PropType = {
  classes: Object,
  fetchOREA: Function,
  testOREADown: Function,
  firstLogin: boolean,
  changeView: (string) => void,
  closeDialog: () => void,
}

export const MakeRequestC = ({
  fetchOREA, classes, testOREADown, firstLogin, changeView, closeDialog,
}: PropType) => (
  <React.Fragment>
    <p className={classes.screenTitle} data-testid="make-request">{c.MAKE_REQUEST_TITLE}</p>
    <div className={classes.notice}>
      <div className={classes.noticeImageContainer}>
        <IllSmallAccess style={{ width: '90px' }} />
      </div>
      <div className={classes.noticeText}>
        {firstLogin
          ? (
            <React.Fragment>
              <span>
                <b>{c.FIRST_LOGIN_MSG_ACCENT}</b>
                {c.FIRST_LOGIN_MSG}
              </span>
            </React.Fragment>
          )
          : c.MAKE_REQUEST_MSG
        }
      </div>
    </div>
    <div className={classes.controlButtons}>
      <Button
        classes={{ button: classes.controlButton }}
        text={firstLogin ? c.ILL_DO_IT_LATER : c.CANCEL}
        onClick={firstLogin ? () => changeView('do-it-later') : () => closeDialog()}
        secondary
      />
      <Button
        testId="fetch-orea"
        classes={{ button: classes.controlButton }}
        text={c.GO_TO_OREA}
        onClick={() => fetchOREA()}
      />
    </div>
    { ENVIRONMENT !== c.ENVIRONMENTS.PROD
      && (
      <Button
        classes={{ button: classes.singleButton }}
        text="***TEST OREA DOWN***"
        onClick={() => testOREADown()}
        fullWidth
      />
      )
    }
  </React.Fragment>
);

export default withStyles(styles)(MakeRequestC);
