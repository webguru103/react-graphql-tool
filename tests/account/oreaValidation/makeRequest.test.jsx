/* eslint import/no-extraneous-dependencies: 0 */

import * as React from 'react';
import 'jest-dom/extend-expect';
import { render, fireEvent, cleanup } from 'react-testing-library';
import { withContext } from '../../test-helpers/contextWrapper';
import { MakeRequestC } from '../../../src/account/oreaValidation/makeRequest';

describe('Make request screen', () => {
  afterEach(cleanup);
  const El = withContext(MakeRequestC);
  it('renders', () => {
    const { container } = render(<El classes={{}} />);
    expect(container).toMatchSnapshot();
  });

  it('attempts to fetch OREA page on click', () => {
    const fetchOREA = jest.fn();
    const { getByTestId } = render(<El classes={{}} fetchOREA={fetchOREA} />);
    const button = getByTestId('fetch-orea');

    fireEvent(
      button,
      new MouseEvent('click', { bubbles: true, cancelable: true }),
    );
    expect(fetchOREA).toHaveBeenCalledTimes(1);
  });

});
