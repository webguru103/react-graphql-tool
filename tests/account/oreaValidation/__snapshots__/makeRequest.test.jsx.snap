// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make request screen renders 1`] = `
<div>
  <p
    data-testid="make-request"
  >
    <span>
      Identity verification
    </span>
  </p>
  <div>
    <div>
      <svg
        style="width: 90px;"
        viewBox="0 0 110 110"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <path
            d="M70.757 63.607c-1.599-.741-2.3-2.259-2.3-3.914V55.22c.348-.366 2.583-3.832 3.666-8.594 1.586-.454 2.775-1.811 2.775-3.425v-5.656c0-1.056-1.051-1.511-1.84-2.171V28.86s.771-10.857-13.802-10.857c-14.572 0-13.802 10.857-13.802 10.857v6.514c-.789.661-1.84 1.116-1.84 2.171v5.646c0 1.264.896 2.388 1.968 3.03 1.295 5.223 4.473 8.697 4.473 8.697L50 58.003c0 1.6-.997 2.25-2.514 3.02-7.36 2.179-12.728 4.351-16.1 6.514-2.479 1.589-3.203 7.253-2.171 16.992 6.632 1.611 16.415 2.446 29.349 2.507 12.842 0 22.664-.78 29.467-2.341a1 1 0 0 0 .771-.874c.775-7.686.128-12.252-1.94-13.698-3.349-2.341-8.716-4.511-16.1-6.509"
            fill="#191b24"
            fill-opacity=".05"
          />
          <path
            d="M73.757 62.607c-1.599-.741-2.3-2.259-2.3-3.914V54.22c.348-.366 2.583-3.832 3.666-8.594 1.586-.454 2.775-1.811 2.775-3.425v-5.656c0-1.056-1.051-1.511-1.84-2.171V27.86s.771-10.857-13.802-10.857c-14.572 0-13.802 10.857-13.802 10.857v6.514c-.789.661-1.84 1.116-1.84 2.171v5.646c0 1.264.896 2.388 1.968 3.03 1.295 5.223 4.473 8.697 4.473 8.697l.214 4.669c0 1.6-.997 3.25-2.514 4.02 0 0-11.04 3.269-16.1 6.514-3.718 2.384-2.44 13.407-2.44 13.407s9.948 2.416 29.349 2.507c19.93 0 30.16-2.507 30.16-2.507s1.331-11.172-1.866-13.407c-5.02-3.512-16.1-6.509-16.1-6.509"
            fill="#7890de"
          />
          <path
            d="M41.805 52.555c-5.687-5.687-15.09-5.392-20.394.884-4.608 5.455-4.207 13.739.873 18.757 3.864 3.817 9.191 4.658 14.01 2.999l2.447 2.497h4.35v3.462h3.462v3.462h4.327v4.327h2.596v2.402c0 .698.034.905.477 1.347.336.336.781.541 1.254.577h6.923v-7.788L43.637 67.853c2.621-5.193 2.502-10.96-1.836-15.298"
            fill="#191b24"
            fill-opacity=".05"
          />
          <path
            d="M42.805 51.285c-5.687-5.687-15.09-5.392-20.394.884-4.608 5.455-4.207 13.739.873 18.757 3.864 3.817 9.191 4.658 14.01 2.999l2.447 2.497h4.35v3.462h3.462v3.462h4.327v4.327h2.596v2.402c0 .698.034.905.477 1.347.336.336.781.541 1.254.577h6.923v-7.788L44.637 66.583c2.621-5.193 2.502-10.96-1.836-15.298m-10.402 9.179a4.328 4.328 0 0 1-6.119-6.12 4.328 4.328 0 0 1 6.119 6.12"
            fill="#e4c05c"
          />
          <path
            d="M42.365 70.365l-4.856 3.777 2.236 2.281h4.35v3.462h3.462v3.462h4.327v4.327h2.596v2.402c0 .698.034.905.477 1.347.336.336.781.541 1.254.577h6.923v-.865L42.365 70.366"
            fill="#dda86f"
          />
          <path
            d="M29.385 51.327a6.06 6.06 0 1 0 0 12.12 6.06 6.06 0 0 0 0-12.12m0 10.385a4.328 4.328 0 0 1 0-8.654 4.328 4.328 0 0 1 0 8.654"
            fill="#bc9c4d"
          />
        </g>
      </svg>
    </div>
    <div>
      <span>
        In order to continue, we need to verify your identity with OREA. Please click the button below to verify your identity.
      </span>
    </div>
  </div>
  <div>
    <button
      class="MuiButtonBase-root-31 MuiButton-root-5 MuiButton-contained-16 MuiButton-containedSecondary-18 ButtonC-secondary-3 MuiButton-raised-19 MuiButton-raisedSecondary-21 ButtonC-button-1"
      data-testid=""
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label-6"
      >
        <span>
          Cancel
        </span>
      </span>
    </button>
    <button
      class="MuiButtonBase-root-31 MuiButton-root-5 MuiButton-contained-16 MuiButton-containedPrimary-17 ButtonC-primary-2 MuiButton-raised-19 MuiButton-raisedPrimary-20 ButtonC-button-1"
      data-testid="fetch-orea"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label-6"
      >
        <span>
          Go to OREA validation page
        </span>
      </span>
    </button>
  </div>
  <button
    class="MuiButtonBase-root-31 MuiButton-root-5 MuiButton-contained-16 MuiButton-containedPrimary-17 ButtonC-primary-2 MuiButton-raised-19 MuiButton-raisedPrimary-20 MuiButton-fullWidth-30 ButtonC-button-1"
    data-testid=""
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label-6"
    >
      ***TEST OREA DOWN***
    </span>
  </button>
</div>
`;
